src/Request/SiteRequestContext.php

```diff
@@ -46,7 +46,11 @@ public function getHost(): string
         $site = $this->getSite();
 
         if ($site && !$site->isLocalhost()) {
-            return $site->getHost();
+            $host = $site->getHost();
+
+            if ($host) {
```

Comment:
I wonder if we should throw an exception when it's null.

Reply:
I went for the exception here.